lib/path.js

```diff
@@ -1,7 +1,14 @@
 'use strict';
 
+const util = require('util');
 const isWindows = process.platform === 'win32';
 
+function assertPath(path) {
+  if (typeof path !== 'string')
+    throw new TypeError('Path must be a string. Received ' +
+                        util.inspect(path));
+}
+
```

Comment:
Thanks, that didn't occur to me of course, because I secretly hate classic `for` loops. I have to jsperf argument iteration sometime, including rest parameters.